### PR TITLE
[TRAFODION-1526]Incorrect messages for ResultSet cursor before the first row and after the last row

### PR DESCRIPTION
**Pull request by kevinxu021**
The fixes are trying to separate the message into 2 pieces(before first row and after last row)

### CODE REVIEW

File: core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/TrafT4ResultSet.java [hunk 1]
```diff
@@ -4299,12 +4299,10 @@ private BaseRow getCurrentRow() throws SQLException {
 			return insertRow_;
 		} else {
 			if (isBeforeFirst_) {
-				String cursorExpt = "The cursor is before the first row, therefore no data can be retrieved.";
-				throw HPT4Messages.createSQLException(connection_.props_, connection_.getLocale(), cursorExpt, null);
+				throw HPT4Messages.createSQLException(connection_.props_, connection_.getLocale(), "cursor_is_before_first_row", null);
```
Comment (DaveBirdsall) [inline core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/TrafT4ResultSet.java]: There was another pull request that changed HPT4Messages to T4Messages. I don't know the details behind it, but I'm wondering if a similar change needs to be made here?

Comment (DaveBirdsall) [inline core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/TrafT4ResultSet.java]: Also, should "cursor_is_before_first_row" be "cursor_is_before_first_row_msg"?

File: core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/TrafT4ResultSet.java [hunk 2]
```diff
@@ -4299,12 +4299,10 @@ private BaseRow getCurrentRow() throws SQLException {
...
 			}
 			if (isAfterLast_) {
-				String cursorExpt = "The cursor is after last row, which could be due to the result set containing no rows, or all rows have been retrieved.";
-				throw HPT4Messages.createSQLException(connection_.props_, connection_.getLocale(), cursorExpt, null);
+				throw HPT4Messages.createSQLException(connection_.props_, connection_.getLocale(), "cursor_after_last_row", null);
```
Comment (DaveBirdsall) [inline core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/TrafT4ResultSet.java]: Similar comments here.

File: core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/TrafT4ResultSet.java (hunk 1 above)
Comment (arvind-narain) [inline core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/TrafT4ResultSet.java]: Hi Dave - change in pr 120 is with the message filename itself.

createSQLException takes in the string supplied and appends "_msg", "_sqlstate", "_sqlcode" to get the corresponding message, state and error code from the message file (now T4Messages)

**Comment by DaveBirdsall** [comment]
Thanks, Kevin, for this change. Thanks, Arvind, for the explanations to my questions. The changes look good to me.

**Comment by sureshsubbiah** [comment]
jenkins, add user

**Comment by arvind-narain** [comment]
Starting the merge. This will need https://github.com/apache/incubator-trafodion/pull/120 (TRAFODION-1525) to work.

**Pull request merged**